### PR TITLE
Array and filter optimizations

### PR DESCRIPTION
**Pull request by banesullivan**
A few internal code path optimizations that I uncovered while benchmarking one of my projects.

### Speed up string array conversions

Two Python loops in `arrays.py` flagged with `OPTIMIZE` markers were burning per-element Python/C crossings.

- `convert_string_array(numpy -> vtk)`: pre-allocate via `SetNumberOfValues(n)` and iterate `arr.tolist()` to skip `numpy.str_` boxing. ~1.4x.
- `_SerializedDictArray._string` setter: store the JSON string as one value instead of one per character. The getter still joins so legacy char-per-value files keep loading. ~85x on a 50-key dict.

### Speed up has array check

`'name' in mesh.point_data` materialized the full key list before the membership test. Replaced with the VTK object's `HasArray`, with a fallback to `keys()` for an edge case for empty-names still resolves.

~76x on `'a5' in point_data` (20 arrays). This adds up when we look at how commonly `if 'foo' in mesh.point_data` occurs.

### DOC: Add performance note about slice `generate_triangles`

See https://github.com/pyvista/pyvista/issues/8256#issuecomment-4265932183

The slice filters defaults to `generate_triangles=False`, which selects `vtkCutter`'s slow per-cell path. `vtkCutter` defaults to `True`:

```text
slice(generate_triangles=False)   118 ms ± 2.3 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
slice(generate_triangles=True)    31.9 ms ± 264 μs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

### Code quality

These aren't really performance gains but just code quality things I noticed while benchmarking mesh creation

### `numpy_to_idarr` ravel

`numpy_to_idarr` always called `ind.ravel()`, even for 1D contiguous input where it only allocates view metadata. Skip it in the 1D case as this is widely used in `vtkCellArray` / `PolyData` / `UnstructuredGrid` construction. 

### Reduce redundant wrappers in `linear_copy`

The quad fix-up loop read `lgrid.cell_connectivity` 4-7 times per branch. Each read allocates a fresh numpy view so changed this to cache it to a local variable. Probably has perf gains for smaller meshes

### CODE REVIEW

File: pyvista/core/utilities/arrays.py [hunk 1]
```diff
@@ -973,14 +978,24 @@ class _SerializedDictArray(DisableVtkSnakeCase, UserDict, _vtk.vtkStringArray):
...
-        self.SetNumberOfValues(0)  # Clear string
-        for char in str_:  # Populate string
-            self.InsertNextValue(char)
+        # Store the entire string as a single value rather than one
+        # character per value. This avoids O(len(str_)) Python<->C
+        # crossings and matches how every other VTK string field stores
+        # text. Reading still works for the legacy char-per-value format
+        # because the getter joins all values.
+        self.SetNumberOfValues(1)
+        self.SetValue(0, str_)
```
Comment (user27182) [inline pyvista/core/utilities/arrays.py:998]: Good catch. Sounds like the original implementation is the result of my naive understanding of how VTK string arrays work.

The docs have a bunch of warnings about `user_dict` being inefficient. Do you think these warnings can be removed?

Comment (banesullivan) [reply pyvista/core/utilities/arrays.py:998]: I changed the docs around them

**Review by user27182 — APPROVED** [review]
